### DISCUSSION TITLE
How to use RequiredMappingStrategy

### DISCUSSION THREAD

**Discussion by R-abodyak** [riok/mapperly#1051] [Q&A]
Hi , I was looking at the docs trying to undurstand diagnostics 

so for this one : https://mapperly.riok.app/docs/api/riok.mapperly.abstractions.requiredmappingstrategy/

I tried the following example , however no warning was emitted 
```

public class Car
{
    public string? Name { get; set; } = string.Empty;
    public CarColor CarColor { get; set; } = CarColor.White;
}

public class CarDto
{
    public string Name { get; set; } = string.Empty;
}

[Mapper(RequiredMappingStrategy = RequiredMappingStrategy.Source)]

public static partial class CarMapper
{
    public static partial CarDto MapCarToDto(Car car);
}

var car = new Car()
{
    Name = "car",
    CarColor = CarColor.Black

};

var carDto = CarMapper.MapCarToDto(car);

```

**Comment by latonz** [comment]
RMG012 and RMG020 do currently have a default severity of info. We may change this in a future release (probably with the next breaking change). To make them visible you need to have an `.editorconfig` with the following rules:
```editorconfig
[*.cs]
dotnet_diagnostic.RMG012.severity = error # Unmapped target member
dotnet_diagnostic.RMG020.severity = error # Unmapped source member
```

See also [strict property mappings](https://mapperly.riok.app/docs/configuration/mapper/#strict-property-mappings)
